src/components/toast/toast.component.tsx

```diff
@@ -131,6 +154,16 @@ export const Toast = React.forwardRef<HTMLDivElement, ToastProps>(
       };
     }, []);
 
+    function iconToRender() {
+      if (isNotification) return "alert";
+      if (variant === "neutral") return "info";
+      if (variant === "success") return "tick_circle";
+      // istanbul ignore if
```

Comment:
I'll DM about this because it's a pain to try and explain over here.

Reply:
What do you think about moving the icon related code from `TypeIcon` into `Toast` itself? As we are currently having to override several parts of functionality from `TypeIcon`  ([lines 214-217](https://github.com/Sage/carbon/pull/6251/files#diff-df110855e4798d403497e65d44d9116a5dfca6a0fb85d146be59e23118a55518R214-R217)), this may make the code in `Toast` easier to read and decouple `Toast` from `Message`.